lua/nvim-treesitter/utils.lua

```diff
@@ -17,6 +17,26 @@ function M.setup_commands(mod, commands)
   end
 end
 
+function M.get_path_sep()
+  return fn.has('win32') == 1 and '\\' or '/'
+end
+
+-- Returns a function that joins the given arguments with separator. Arguments
+-- can't be nil. Example:
+--[[
+print(M.generate_join(" ")("foo", "bar"))
+--]]
+-- prints "foo bar"
+function M.generate_join(separator)
+  return function (...)
+    return table.concat({...}, separator)
+  end
+end
+
+M.join_path = M.generate_join(M.get_path_sep())
+
+M.join_space = M.generate_join(" ")
```

Comment:
I think we can have all of them as `local function`, to avoid spilling to much things out.